x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/results/SingleValueInferenceResults.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.core.ml.inference.results;
+
+import org.elasticsearch.common.ParseField;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+
+import java.io.IOException;
+
+public abstract class SingleValueInferenceResults implements InferenceResults<Double> {
```

Comment:
I am not sure this helps more than it confuses.

- its name is misleading as `value` is more generic than `double`
- it means classes that inherit this have to call their value field `value`. Not sure this is the case.

Reply:
> its name is misleading as `value` is more generic than `double`

I disagree. `double` is exactly the numeric value we need. A whole `double` would be returned via classification, and any `double` could be returned via regression. 

> it means classes that inherit this have to call their value field `value`. Not sure this is the case.

Could you provide a counter example?